### DISCUSSION TITLE
Graph embedding network

### DISCUSSION THREAD

**Discussion by ilyalasy** [pyg-team/pytorch_geometric#2549] [Q&A] [answered]
Hey, would love to hear your advice. What would be the best choice of network for graph embedding that can encapsulate it's features in a "good" way? I know the question is very general, maybe I should rephrase it: What would be the criteria for choosing specific graph embedding network ?

**Comment by rusty1s** [comment, marked as answer]
It's hard to give a general advice on this one. However, there exists GNNs that are more suitable for a given task than others., e.g.:
* If you think your problem highly depends on the structure of the graph, you should go for more structured-focused/expressive GNNs such as `GIN` or `PANConv`.
* If you think your problem is more feature-based, e.g., which features of neighbors point to which other features, `GATConv` or `TransformerConv` is a great choice.
* If you think that "smoothing" features is sufficient for your problem, `GCNConv` or `SAGEConv` might be the way to go.
* If you have rich edge feature information, you need a GNN that can utilize those, e.g., `NNConv` or `GMMConv`.
* If you think that your problem is inherently hierarchical, you may want to try out different pooling approaches like `DiffPool`.

**Reply by ilyalasy** [reply]
Thanks for this guidelines, they are pretty useful!
